File: cine-app/src/app/core/services/pelicula.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Pelicula } from '../models/pelicula.model';
import { environment } from '../../Environments/environment';

@Injectable({
  providedIn: 'root'
})
export class PeliculaService {
  private apiUrl = environment.apiUrl + '/Pelicula';

  constructor(private http: HttpClient) {}

  obtenerTodas(): Observable<Pelicula[]> {
    return this.http.get<Pelicula[]>(this.apiUrl);
  }
  obtenerPorId(id: number): Observable<Pelicula> {
    return this.http.get<Pelicula>(`${this.apiUrl}/${id}`);
  }
  crear(pelicula: Pelicula): Observable<Pelicula> {
    return this.http.post<Pelicula>(this.apiUrl, pelicula);
  }

  actualizar(id: number, pelicula: Pelicula): Observable<Pelicula> {
    return this.http.put<Pelicula>(`${this.apiUrl}/${id}`, pelicula);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: CineAPI/CineAPI/Services/PeliculaService.cs
using CineAPI.Model;
using CineAPI.Repository.Interfaces;
using CineAPI.Services.Interfaces;

namespace CineAPI.Services
{
    public class PeliculaService : IPeliculaService
    {
        private readonly IPeliculaRepository _repo;

        public PeliculaService(IPeliculaRepository repo)
        {
            _repo = repo;
        }

        public async Task<List<Pelicula>> ObtenerTodas()
        {
            return await _repo.ObtenerTodas();
        }

        public async Task<Pelicula?> ObtenerPorId(int id)
        {
            return await _repo.ObtenerPorId(id);
        }

        public async Task Crear(Pelicula pelicula)
        {
            await _repo.Crear(pelicula);
        }

        public async Task Actualizar(Pelicula pelicula)
        {
            await _repo.Actualizar(pelicula);
        }

        public async Task Eliminar(int id)
        {
            await _repo.Eliminar(id);
        }
    }
}